dataplane/forwarding/fwdport/ports/kernel.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ports
+
+import (
+	"fmt"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcap"
+	"github.com/openconfig/lemming/dataplane/forwarding/fwdaction"
+	"github.com/openconfig/lemming/dataplane/forwarding/fwdport"
+	"github.com/openconfig/lemming/dataplane/forwarding/infra/fwdcontext"
+	"github.com/openconfig/lemming/dataplane/forwarding/infra/fwdobject"
+	"github.com/openconfig/lemming/dataplane/forwarding/infra/fwdpacket"
+
+	log "github.com/golang/glog"
+	fwdpb "github.com/openconfig/lemming/proto/forwarding"
+)
+
+func init() {
+	fwdport.Register(fwdpb.PortType_PORT_TYPE_KERNEL, kernelBuilder{})
+}
+
+type kernelBuilder struct {
+}
```

Comment:
a short comment please. I think my main confusion when starting to read this is what a "kernel" port is in the context of this code. From the proto it looks like it's a linux network device identified by a name. Would be nice to see a comment for this so it's clear from the outset.